tests/internal/e2e/run_e2e_tests.sh

```diff
@@ -1,13 +1,76 @@
 #!/bin/bash
 ROOT=$(dirname "$(realpath "$0")")
 
+# Get the directory of the current script
+SCRIPT_DIR=$(dirname "$0")
+
+# Define the path to the specs file relative to the script directory
+SPEC_PATH="$SCRIPT_DIR/test/specs_config.ts"
+
+# Check if the specs file exists
+if [ ! -f "$SPEC_PATH" ]; then
+  echo "The file $SPEC_PATH does not exist."
+  exit 1
+fi
+
+# Compile the specs file 
+npx tsc "$SPEC_PATH"
+
+# Path to the compiled JavaScript file
+JS_FILE="./test/specs_config.js"
+
 activate_24_release () {
   export VECTORCAST_DIR=/vcast/release24
   export PATH=/vcast/release24:$PATH
   export ENABLE_ATG_FEATURE=TRUE
   echo "Vcast 24 is activated"
 }
 
+set_specs_params() {
+  # Check if RUN_GROUP_NAME is set
+  if [ -z "$RUN_GROUP_NAME" ]; then
+    echo "RUN_GROUP_NAME is not set. Please set it and try again."
+    exit 1
+  fi
+
+  # Check if the compiled JavaScript file exists
+  if [ ! -f "$JS_FILE" ]; then
+    echo "Compiled JavaScript file not found! Please compile the TypeScript file first."
+    exit 1
+  fi
+
+  # Extract environment variables for the given group using the compiled JavaScript file
+  env_vars=$(node -e "
+    const { getEnvVarsForGroup } = require('$JS_FILE');
+    const groupName = process.env.RUN_GROUP_NAME;
+    console.log(groupName)
+    const envVars = getEnvVarsForGroup(groupName);
+    if (envVars) {
+      console.log(envVars);
+    } else {
+      console.error('No environment variables found or group not found.');
+      process.exit(1); // Exit with an error code if no variables found
+    }
+  ")
+
+  # Check if env_vars is empty, indicating either the group or env section might be missing
+  if [ -z "$env_vars" ]; then
+    echo "Spec group $RUN_GROUP_NAME not found or has no environment variables."
+    exit 1
+  fi
+
+  # Export each environment variable
+  while IFS= read -r line; do
+    export "$line"
+  done <<< "$env_vars"
```

Comment:
I feel this might be overly complicated for what it needs to be. Why don't we just have a simple `for`?
Something along these lines:
```sh
for var_name in $env_vars; do
  export "$var_name"
done
```

Reply:
`var_name` is wrong, because it is likely `env_var_val`, right?

Reply:
Adressed in https://github.com/vectorgrp/vector-vscode-vcast/commit/37b4bb4951ced74173e6f06790b959641bd4d1c2 and deleted hardocded exports in https://github.com/vectorgrp/vector-vscode-vcast/commit/97e6eb7d71236b35712a55095f21026478d6dff6.